sklearn/calibration.py

```diff
@@ -589,14 +589,13 @@ def calibration_curve(y_true, y_prob, normalize=False, n_bins=5,
     if strategy == 'quantile':  # Determine bin edges by distribution of data
         quantiles = np.linspace(0, 1, n_bins + 1)
         bins = np.percentile(y_prob, quantiles * 100)
-        bins[-1] = bins[-1] + 1e-8
     elif strategy == 'uniform':
-        bins = np.linspace(0., 1. + 1e-8, n_bins + 1)
+        bins = np.linspace(0., 1., n_bins + 1)
     else:
         raise ValueError("Invalid entry to 'strategy' input. Strategy "
                          "must be either 'quantile' or 'uniform'.")
 
-    binids = np.digitize(y_prob, bins) - 1
+    binids = np.searchsorted(bins[1:-1], y_prob, side='right')
```

Comment:
maybe add a comment which says that we use side="right" to keep backward compatibility.

Reply:
I feel like that comment would become obscure very quickly? I'm not sure how it'll help in the future.